### PR TITLE
mainnet_v1: bump to Godwoken/Web3 Web3-Indexer v1.12.0 

### PR DESCRIPTION
**Pull request by Simon-Tl**
## [Notable Changes](https://github.com/godwokenrises/godwoken/blob/develop/CHANGELOG.md)
- Builtin mainnet consensus config https://github.com/godwokenrises/godwoken/pull/946
  The consensus-related options were moved into consensus config, and the builtin mainnet consensus were put into the godwoken binary. This change reduces the operation works of node maintainers
- perf: optional SMT trie feature and migrate command https://github.com/godwokenrises/godwoken/pull/859
- feat: support CKB built-in indexer https://github.com/godwokenrises/godwoken/pull/907
- Refactor rpc server https://github.com/godwokenrises/godwoken/pull/927
- Improve the error code of API when executing transactions https://github.com/godwokenrises/godwoken/pull/930
- gwos-evm: fix big mod exp https://github.com/godwokenrises/godwoken/pull/932
- fix(web3): eth_getFilterLogs should return all matched logs https://github.com/godwokenrises/godwoken/pull/947
- fix(withdrawal_unlocker): remove dry_run_transaction and fix error display https://github.com/godwokenrises/godwoken/pull/992
- https://github.com/godwokenrises/godwoken/pull/1000

## Detailed Release notes
   - Godwoken
      https://github.com/godwokenrises/godwoken/releases

## Changed Configs
0. support CKB build-in indexer
If a CKB indexer RPC url is explicitly specified, we assume that we are connecting to a standalone indexer.
Otherwise we just use CKB RPC url and assume that we are connecting to CKB builtin indexer.

2.  introduced mainnet builtin consensus config
   ```TOML
    # builtin consensus config
    # see https://github.com/godwokenrises/godwoken/pull/946
    [consensus]
    # https://github.com/godwokenrises/godwoken/blob/develop/crates/config/src/consensus/builtins/mainnet.toml
    builtin = 'Mainnet'
   ```

3. [[dynamic_config.fee_config] -> [mem_pool.fee]](https://github.com/godwokenrises/godwoken-info/pull/89/files#:~:text=%5Bmem_pool.fee,withdraw_cycles_limit%20%3D%2020000)

4. [[dynamic_config.rpc_config] -> [mem_pool.extra]](https://github.com/godwokenrises/godwoken-info/pull/89/files#:~:text=withdraw_cycles_limit%20%3D%2020000-,%5Bdynamic_config.rpc_config%5D,sudt_proxy_code_hashes%20%3D%20%5B%5D,-Footer)

### CODE REVIEW

File: mainnet_v1/docker-compose.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ volumes:
 services:
   gw-readonly:
     container_name: gw-mainnet_v1-readonly
-    image: ghcr.io/godwokenrises/godwoken:1.7.4-poly.1.5.1
+    image: ghcr.io/godwokenrises/godwoken:v1.12.0
```
Comment (Flouse) [inline mainnet_v1/docker-compose.yml:11]: Please update the images of web3 and web3-indexer too.

**Review by Flouse — CHANGES_REQUESTED** [review]

File: mainnet_v1/gw-mainnet_v1-config-readonly.toml [hunk 2]
```diff
@@ -180,80 +58,18 @@ max_cycles_limit = '1950000000'
...
 meta_cycles_limit = 20000
 sudt_cycles_limit = 20000
 eth_addr_reg_cycles_limit = 20000
 withdraw_cycles_limit = 20000
 
-[dynamic_config.rpc_config]
+[mem_pool.extra]
 allowed_sudt_proxy_creator_account_id = [3]
-sudt_proxy_code_hashes = []
+sudt_proxy_code_hashes = []
```
Comment (Flouse) [inline mainnet_v1/gw-mainnet_v1-config-readonly.toml]:
```suggestion
sudt_proxy_code_hashes = ["0xde4542f5a5bd32c09cd98e9752281f88900a059aab7ac103edd9df214f136c52"]

```

**Review by Flouse — CHANGES_REQUESTED** [review]

File: mainnet_v1/gw-mainnet_v1-config-readonly.toml (hunk 2 above)
Comment (Simon-Tl) [reply mainnet_v1/gw-mainnet_v1-config-readonly.toml]: I see the readonly nodes ```sudt_proxy_code_hashes```  is empty

Comment (Flouse) [reply mainnet_v1/gw-mainnet_v1-config-readonly.toml]: yes, it was ok.
`sudt_proxy_code_hashes` is an optional config on a gw-readonly-node, while it is an important config on gw-fullnode.

Let's make this config clear in a gw-readonly-node too.

## Related Code
### Only accounts in allow list could create sUDT proxy contract.
https://github.com/godwokenrises/godwoken/blob/118163d10fd62761463af8c788f32aa23c98e365/crates/tx-filter/src/erc20_creator_allowlist.rs#L22-L50

Comment (Simon-Tl) [reply mainnet_v1/gw-mainnet_v1-config-readonly.toml]: OK

**Review by Flouse — COMMENTED** [review]
Now, godwoken:v1.12.0 can use ckb-builtin-indexer.

**Review by Flouse — APPROVED** [review]

File: mainnet_v1/gw-mainnet_v1-config-readonly.toml [hunk 3]
```diff
@@ -144,29 +38,17 @@ enable_debug_rpc = false
...
-[block_producer.challenger_config.burn_lock]
-code_hash = '0x0000000000000000000000000000000000000000000000000000000000000000'
-hash_type = 'data'
-args = '0x'
-
 [mem_pool]
-execute_l2tx_max_cycles = 3000000000
-restore_path = '/mnt/mem_block'
-
+execute_l2tx_max_cycles = 500000000
```
Comment (jjyr) [inline mainnet_v1/gw-mainnet_v1-config-readonly.toml]: Why change this value?

Comment (Flouse) [reply mainnet_v1/gw-mainnet_v1-config-readonly.toml]:
```suggestion
execute_l2tx_max_cycles = 3000000000
```

Good catch.
It should be 3000000000, the same value as Godwoken fullnode.

**Review by Flouse — CHANGES_REQUESTED** [review]

**Review by Flouse — APPROVED** [review]

**Review by jjyr — APPROVED** [review]

**Pull request merged**